spyder/config/main.py

```diff
@@ -180,7 +185,8 @@
               'truncate': True,
               'minmax': False,
               'show_callable_attributes': True,
-              'show_special_attributes': False
+              'show_special_attributes': False,
+              'filter_on': False
```

Comment:
The filter needs to be on by default (contrary to what was mentioned on issue https://github.com/spyder-ide/ux-improvements/issues/17). Sorry if that made you do more work than necessary, probably we got confused when we discussed about it some years ago.

That's because if the filter is not on, then the Variable Explorer would show too many unnecessary variables to users (like private IPython references that users don't need to touch, functions and modules).